### PR TITLE
Serial Print, LCD support, and while loop

### PR DESCRIPTION
**Pull request by stevoleeto**
This branch includes the following changes from the master branch:

- New Serial.print which prints to the NEXT line instead of tabbing to the right. This is much easier to read and is more intuitive to the average user.

- LCD support for LCD's such as: https://www.sparkfun.com/products/709. These LCD's are common in most beginner's kits, and are widely used. This includes three new blocks; Setup, Print, and Clear.

- While Loop. A block for creating a while loop.

### CODE REVIEW

**Comment by Arachnid** [comment]
Please forgive the unsolicited review; obviously this isn't my repo, so nothing I say has any weight at all except as a suggestion. :)

File: blockly/blocks/base.js [hunk 1]
```diff
@@ -196,3 +196,56 @@ Blockly.Blocks['serial_print'] = {
     this.setTooltip('Prints data to the console/serial port as human-readable ASCII text.');
   }
 };
+
+Blockly.Blocks['lcd_setup'] = {
+  init:function(){
```
Comment (Arachnid) [inline blockly/blocks/base.js]: Code across various branches will be a lot easier to read if we keep to a consistent coding style - including putting spaces between 'init' and 'function, indenting continued lines ('.appendField' etc).

File: blockly/blocks/base.js [hunk 2]
```diff
@@ -196,3 +196,56 @@ Blockly.Blocks['serial_print'] = {
...
+  }
+};
+
+Blockly.Blocks['lcd_print'] = {
+  init:function(){
+    this.setColour (257);
+    this.appendDummyInput("")
+    .appendField("LCD Print")
+    .appendField("Cursor Column #: ")
+    .appendField(new Blockly.FieldDropdown([["1", "0"], ["2", "1"], ["3", "2"], ["4", "3"], ["5", "4"], ["6", "5"], ["7", "6"], ["8", "7"], ["9", "8"], ["10", "9"], ["11", "10"], ["12", "11"],["13", "12"],["14", "13"],["15", "14"],["16", "15"]]), "CURSOR_COLUMN#")
```
Comment (Arachnid) [inline blockly/blocks/base.js]: Could you generate this array instead of enumerating it?

File: blockly/generators/arduino/control.js [hunk 3]
```diff
@@ -75,3 +75,20 @@ Blockly.Arduino.controls_for = function() {
   }
   return code;
 };
+
+Blockly.Arduino.controls_whileUntil = function(){
```
Comment (Arachnid) [inline blockly/generators/arduino/control.js]: Did you leave out the block definition for this?

**Comment by stevoleeto** [comment]
@Arachnid never :) I just started doing this stuff in January, so any feedback is always welcome.

File: blockly/blocks/base.js (hunk 1 above)
Comment (stevoleeto) [inline blockly/blocks/base.js]: I completely agree with this. It obviously slipped here and I'll make sure next time to follow these more closely!

File: blockly/blocks/base.js (hunk 2 above)
Comment (stevoleeto) [inline blockly/blocks/base.js]: Hmm note sure exactly what you mean. When I did this in January I had no JavaScript experience.  I assume generating the array would take place in the Arduino.js file with the other arrays?

File: blockly/generators/arduino/control.js (hunk 3 above)
Comment (stevoleeto) [inline blockly/generators/arduino/control.js]: Nope! Surprisingly the block definition was already made for this. Take a peek inside of blocks/loop.js. It should be the third one.

Comment (Arachnid) [inline blockly/generators/arduino/control.js]: Ah, awesome, my mistake.

File: blockly/blocks/base.js (hunk 2 above)
Comment (Arachnid) [inline blockly/blocks/base.js]: You could do that, but you could also just generate it in-place in this function:
columns = [];
for(i = 0; i < 16; i++) columns.append([(i + 1).toString(), i.toString()]);

Comment (stevoleeto) [inline blockly/blocks/base.js]: Ah gotcha, I see what you mean. I will change this when I get a chance!

**Comment by gasolin** [comment]
please put each feature in each commit. You could do it by 

```
$ git branch <branch-name>
$ git cherry-pick <commit hash>
$ git push origin <branch-name>
```

**Comment by stevoleeto** [comment]
@gasolin each feature in a commit? I'm assuming you meant branch. I'll do ahead and do that.

**Comment by Arachnid** [comment]
@stevoleeto I think he meant that he wants you to squash your commits on this branch so there's exactly one commit related to each new feature.

**Comment by stevoleeto** [comment]
@Arachnid so get rid of stepper motor, shift register, revert stepper motor and revert shift register commits correct? Then I'll have just the three commits with their own feature.

**Comment by Arachnid** [comment]
That sounds right to me - assuming that's what gasolin meant, anyway.

**Comment by stevoleeto** [comment]
@Arachnid @gasolin Okay just removed the other commits. Now we have just those three commits for the three features. 

Sorry for being so dumb at this stuff, first time contributing online. You guys have been very patient and helpful.

**Comment by gasolin** [comment]
I mean if there are 3 features you want merge to blocklyduino, it's better to provide 3 separate pull requests. The reason is focus on one feature at a time could be easier to review and identify issues.

**Comment by stevoleeto** [comment]
@gasolin That makes sense to me. I'll go ahead and make three new PR's. You can discard this one

**Pull request closed without merging**